app/cdc_uart_bridge.c

```diff
@@ -0,0 +1,124 @@
+#include <FreeRTOS.h>
+#include <task.h>
+#include <queue.h>
+
+#include "cdc_uart_bridge.h"
+#include "drivers/max14662.h"
+#include "drivers/uart.h"
+#include "drivers/usb/cdc.h"
+#include "stm32l4xx_hal.h"
+
+#define RX_TASK_STACK_SIZE		512
+#define RX_TASK_NAME			"CDCrx"
+#define RX_TASK_PRIORITY		1
+
+#define TX_TASK_STACK_SIZE		512
+#define TX_TASK_NAME			"CDCtx"
+#define TX_TASK_PRIORITY		1
+
+#define QUEUE_LENGTH			10
+#define QUEUE_ITEM_SIZE			sizeof(struct rx_queue_item)
+
+/* Fun test string: 
+</////////////////////////////////////////////////////////////>[##################]
+<2345678901234567890123456789012345678901234567890123456789012>[ABCDEFGHIJKLMNOPQR]
+*/
+
+static struct {
+	bool initialized;
+	StackType_t rx_task_stack[RX_TASK_STACK_SIZE];
+	TaskHandle_t rx_task_handle;
+	StaticTask_t rx_task_tcb;
+	StackType_t tx_task_stack[RX_TASK_STACK_SIZE];
+	TaskHandle_t tx_task_handle;
+	StaticTask_t tx_task_tcb;
+	StaticQueue_t tx_queue;
+	QueueHandle_t tx_queue_handle;
+	uint8_t tx_queue_storage[QUEUE_LENGTH * QUEUE_ITEM_SIZE];
+} self;
+
+static void rx_task(void *p_arg)
+{
+	struct rx_queue_item rx_queue_item;
+	err_t r;
+
+	for (;;) {
+		r = usb_cdc_rx(&rx_queue_item, portMAX_DELAY);
+		if (r != ERR_OK) {
+			while (1)
+				;
+		}
+		r = uart_tx(rx_queue_item.data, rx_queue_item.len, portMAX_DELAY, true);
+		if (r != ERR_OK) {
+			while (1)
+				;
+		}
+	}
+}
+
+static void tx_task(void *p_arg)
+{
+	err_t r;
+	struct rx_queue_item item;
+
+	for (;;) {
+		/* uart_start_rx starts a continuous DMA transfer of 64 bytes that sends its data to our 
+		 * queue that we receive here. In order to get shorter messages in near realtime we need
+		 * a timeout and flush the received bytes so far. */
+		if (xQueueReceive(self.tx_queue_handle, &item, 250) == pdFALSE) {
```

Comment:
self.tx_queue_handle hmm. confusing name now without a prefix. it's the tx task's queue but it is receiving uart data